api/tests/Feature/NotificationTest.php

```diff
@@ -0,0 +1,182 @@
+                    ... on PoolCandidateStatusChangedNotification {
+                        oldStatus
+                        newStatus
+                    }
+                }
+            }
+        }
+    ';
+
+    protected function setUp(): void
+    {
+
+        parent::setUp();
+
+        $this->seed(ClassificationSeeder::class);
+        $this->seed(GenericJobTitleSeeder::class);
+        $this->seed(SkillFamilySeeder::class);
+        $this->seed(SkillSeeder::class);
+        $this->seed(RolePermissionSeeder::class);
+
+        $this->user = User::factory()
+            ->asApplicant()
+            ->create();
+
+        $this->pool = Pool::factory()
+            ->published()
+            ->create();
+
+        $this->originalStatus = PoolCandidateStatus::NEW_APPLICATION->name;
+        $this->newStatus = PoolCandidateStatus::PLACED_TERM->name;
+        $this->user->notify(
+            new PoolCandidateStatusChanged(
+                $this->originalStatus,
+                $this->newStatus,
+                $this->pool->id,
+                $this->pool->name
+            )
+        );
+
+        $this->notification = $this->user->unreadNotifications()->first();
+    }
+
+    public function testQueryNotification(): void
+    {
+
+        $this->actingAs($this->user, 'api')
+            ->graphQL($this->queryNotifications)->assertJson([
+                'data' => [
+                    'notifications' => [
+                        'data' => [
+                            [
+                                'id' => $this->notification->id,
+                                'readAt' => null,
+                            ],
+                        ],
+                    ],
+                ],
+            ]);
+    }
+
+    public function testUnreadNotification(): void
+    {
+        $response = $this->actingAs($this->user, 'api')
+            ->graphQL(/** @lang GraphQL */ '
+                mutation unReadNotification($id: UUID!) {
+                    markNotificationAsUnread(id: $id) {
+                        id
+                        readAt
+                    }
+                }
+            ', ['id' => $this->notification->id]);
+
+        $readAt = $response->json('data.markNotificationAsRead.readAt');
+
+        $this->assertNull($readAt);
+    }
+
+    public function testDeleteNotification(): void
+    {
+        $this->actingAs($this->user, 'api')
+            ->graphQL(/** @lang GraphQL */ '
+                mutation DeleteNotification($id: UUID!) {
+                    deleteNotification(id: $id) {
+                        id
+                    }
+                }
+            ', ['id' => $this->notification->id])->assertJson([
+                'data' => [
+                    'deleteNotification' => [
+                        'id' => $this->notification->id,
+                    ],
+                ],
+            ]);
+
+        $this->assertDatabaseMissing('notifications', ['id' => $this->notification->id]);
```

Comment:
Actually, I'd like to make the delete mutation a soft-delete.

Reply:
Updated to be soft deletes. [soft delete notifications](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9694/commits/fd2f7757e5d0e82478f0ed6990dbd4792b0a74e2)